src/main/java/com/squadb/workassistantapi/rental/domain/Rental.java

```diff
@@ -58,13 +61,39 @@ private Rental(RentalStatus status, LocalDateTime startDate, LocalDateTime endDa
         this.book = book;
     }
 
-    public static Rental createRental(Book book, Member member, boolean isLongTerm, LocalDateTime rentalStartDate,
-                                      RentalValidator rentalValidator, ReservationFinisher reservationFinisher) {
-        rentalValidator.validateNotExistsOtherMemberReservation(book, member);
-        reservationFinisher.finish(book, member);
+    public static Rental createRental(Book book, Member member, List<Reservation> reservations, boolean isLongTerm, LocalDateTime rentalStartDate) {
+        validateExistsOnlyMemberReservation(book, member, reservations);
         book.decreaseStock();
-        final LocalDateTime endDate = isLongTerm ? null : rentalStartDate.plusDays(NORMAL_RENTAL_DAYS);
-        return new Rental(ON_RENTAL, rentalStartDate,endDate, member, book);
+        finishBookReservation(book, member, reservations);
+        return new Rental(ON_RENTAL, rentalStartDate, calculateEndDate(isLongTerm, rentalStartDate), member, book);
+    }
+
+    private static void validateExistsOnlyMemberReservation(Book book, Member member, List<Reservation> reservations) {
+        if (CollectionUtils.isEmpty(reservations)) {
+            return;
+        }
+
+        long reservationCountByMemberAndBook = reservations.stream()
+                .filter(reservation -> reservation.isWaitingBy(member, book))
+                .count();
+
+        if ((long) reservations.size() != reservationCountByMemberAndBook) {
+            throw new IllegalStateException("다른 고객이 예약중이라 대여할 수 없습니다.");
+        }
+    }
+
+    private static LocalDateTime calculateEndDate(boolean isLongTerm, LocalDateTime rentalStartDate) {
+        return isLongTerm ? null : rentalStartDate.plusDays(NORMAL_RENTAL_DAYS);
+    }
+
+    private static void finishBookReservation(Book book, Member member, List<Reservation> reservations) {
+        List<Reservation> reservationsByMemberAndBook = reservations.stream()
```

Comment:
아래처럼 stream에서 모든 연산을 끝낼 수 있을 거 같아요!

```suggestion
        reservations.stream()
                .filter(reservation -> reservation.isWaitingBy(member, book))
                .forEach(reservation -> reservation.finishedBy(member));
```

Reply:
오 stream으로 다 끝낼 수 있네요 👍